### ISSUE TITLE
Profile parsing problem

### ISSUE THREAD

**Issue by NingyuShi** [conan-io/conan#606]
In [profile.py](https://github.com/conan-io/conan/blob/b5d2cd4e99bfcdea4624c776fd7892acf72a26e0/conans/model/profile.py#L32), the env doesn't work if the value contains a '=', the code is using simple split which does not cover this case.

``` python
        for env in doc.env.split("\n"):
            if env:
                varname, value = env.split("=")
                obj.env[varname] = value
```

Also when an error happens when parsing the profile file, it reports error as cannot find this profile, and the message looks like

```
RROR: Specified profile 'gcc62' doesn't exist.
Existing profiles: gcc62, gcc4.8.3
```

which is confusing.

**Comment by memsharded** [comment]
The second issue has been already fixed and merged to develop: https://github.com/conan-io/conan/pull/585. It will be released in next conan 0.15

The other indeed can be fixes, most likely by splitting only on the first "=", leaving the remaining. 
Just some curiosity, which env var are you using that contains an "="?

**Comment by lasote** [comment]
I think that the commit was released in the last minor versión 0.14.1 so it can be an additional problem. I'll fix it with the "=" issue and will be available in the next release.

**Comment by memsharded** [comment]
Checked, you are right, it was released in 0.14.1.

The issue is that such parsing problem is not covered:

```python
if "=" not in env:
       raise ConanException("Invalid env line '%s'" % env)
varname, value = env.split("=")
obj.env[varname.strip()] = value.strip()
```

A try-except that cover the full parsing method would be useful to capture all errors.

**Comment by NingyuShi** [comment]
@memsharded The case is
```
CXX_FLAGS="-DAAA=0"
```

**Comment by memsharded** [comment]
Fixed in #608, will be released in next 0.15
